platform-sdk/swirlds-state-impl/src/main/java/com/swirlds/state/merkle/MerkleStateRoot.java

```diff
@@ -830,7 +830,9 @@ public void commit() {
          * @param stateKey the state key
          */
         public void remove(String stateKey) {
-            stateMetadata.remove(stateKey);
+            if (!Map.of().equals(stateMetadata)) {
+                stateMetadata.remove(stateKey);
```

Comment:
Why do we need this ?

Reply:
It's necessary because `MerkleStateRoot.getWritableStates()` [here](https://github.com/hashgraph/hedera-services/blob/main/platform-sdk/swirlds-state-impl/src/main/java/com/swirlds/state/merkle/MerkleStateRoot.java#L265) returns the immutable `Map.of()` singleton for a service that has not registered any states; and this is an instance of `ImmutableMap` that throws `UnsupportedOperationException` on receiving `remove()`.

I can go into more detail if you want; but either this or replacing `Map.of()` with `new HashMap<>()` is needed.